crates/bevy_ecs/src/storage/resource.rs

```diff
@@ -175,16 +209,26 @@ impl Resources {
     ) -> &mut ResourceData {
         self.resources.get_or_insert_with(component_id, || {
             let component_info = components.get_info(component_id).unwrap();
+            self.component_ids.push(component_id);
             ResourceData {
-                column: Column::with_capacity(component_info, 1),
+                // SAFETY: component_info.drop() is valid for the types that will be inserted.
```

Comment:
Can you rewrite this to explain how we know that it's valid?

Reply:
Done.